google-http-client/src/main/java/com/google/api/client/http/GenericUrl.java

```diff
@@ -99,24 +102,58 @@ public GenericUrl() {}
    * @throws IllegalArgumentException if URL has a syntax error
    */
   public GenericUrl(String encodedUrl) {
-    this(parseURL(encodedUrl));
+    this(encodedUrl, false);
+  }
+
+  /**
+   * Constructs from an encoded URL.
+   *
+   * <p>Any known query parameters with pre-defined fields as data keys will be parsed based on
+   * their data type. Any unrecognized query parameter will always be parsed as a string.
+   *
+   * <p>Any {@link MalformedURLException} is wrapped in an {@link IllegalArgumentException}.
+   *
+   * <p>Upgrade warning: starting in version 1.18 this parses the encodedUrl using new
```

Comment:
We don't need to reference earlier versions though the info about limited set of schemes is relevnt

Reply:
Not sure I understand the point. Again I am overloading a constructor and this is the original javadoc. Maybe those concerns should be raised in a different issue?

Reply:
This is a new method with new API doc. No reason to copy outdated content form older methods.

Reply:
Done